rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -688,8 +789,17 @@ extern "C" rmw_node_t * rmw_create_node(
 #endif
 
   dds_qos_t * qos = dds_create_qos();
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
+    security_options, "rmw_create_node: Unable to create qos", return nullptr);
```

Comment:
I suspect this was meant to check ``qos`` rather than ``security_options``; and if bailing out, a domain will have been instantiated/refcounted already by ``check_create_domain_locked`` and, as noted, its work must be undone using ``node_gone_from_domain_locked`` (see a few lines below).

Reply:
Fixed with 1ca2269, thanks for catching!